### DISCUSSION TITLE
Difference between G_0 and D_0?

### DISCUSSION THREAD

**Discussion by Mangio621** [voicepaw/so-vits-svc-fork#480] [Q&A] [answered]
Hello.

I've noticed that for each generation, there's a "G" and a "D" version. I know that using "G" is correct and sounds good and D sounds weird on inference. So I know the point of the "G" version... But what's the point for the "D" version? Is this only for training purposes? (Used only for training the model?).

**Comment by xieyumc** [comment]
Yes, the "D" is designed for judge the "G".So it used only on training.

**Comment by Qualzz** [comment, marked as answer]
D is for Discriminator, G is for Generator. 
Basically the Generator model learn how to create the best voice, the D job is to detect if it's a fake voice generated by G or a real sample. 
If G does a good job and D got tricked, D will be better at differentiating fake from real samples. If G produce a sample not convincing enough for D, G will be better at generating a fake voice. 

Basically both models are learning from each others during training.
